examples/http/bulk_ssl_certificate.js

```diff
@@ -0,0 +1,144 @@
+ * @returns Promise that wait for the https call to the targetServer and parse the needed data
+ */
+function getCertificateData(targetServer) {
+    var d = D.q.defer();
+    var website = D.createExternalDevice(targetServer);
+    website.http.get(
+        {
+            url: "/",
+            headers: {
+                connection: "keep-alive",
+            },
+            rejectUnauthorized: false,
+            protocol: "https"
+        }, function (err, resp) {
+            if (err) {
+                console.error(err);
+                D.failure();
+            }
+            var data = null;
+            if(resp && resp.connection && resp.connection.getPeerCertificate){
+                var cert = resp.connection.getPeerCertificate();
+                if(cert && Object.keys(cert).length)
+                    data = {
+                        server: targetServer,
+                        issuer: cert.issuer.O,
+                        expiry: cert.valid_to,
+                        valid: !resp.connection.authorizationError,
+                        certError: resp.connection.authorizationError
+                    };
+            } 
+            d.resolve(data);
+        });
+
+    return d.promise;
+}
+
+/**
+ * 
+ * @param {*} dataList https parsed data for each server in the serversToCheck
+ * @returns same dataList with added remainingDays attribute for each data in the list
+ */
+function parseDates(dataList) {
+    return dataList.map(parseDate);
+}
+
+
+/**
+ * 
+ * @param {*} data contains the data passed by getCertificateData function to calculate remaining days for the certificate
+ * @returns the same data in the input with added remainingDays attribute
+ */
+function parseDate(data) {
+    if(!data) return null;
+    var expiryParsed = data.expiry.match(/^(...) (..) (..):(..):(..) (....) GMT$/);
+    var month = months.indexOf(expiryParsed[1]);
+    var day = expiryParsed[2];
+    var hour = expiryParsed[3];
+    var min = expiryParsed[4];
+    var sec = expiryParsed[5];
+    var year = expiryParsed[6];
+    var date = new Date();
+    date.setUTCFullYear(year, month, day);
+    date.setUTCHours(hour, min, sec, 0);
+    var diff = Math.floor((date.getTime() - new Date().getTime()) / (1000 * 60 * 60 * 24));
+    data.remainingDays = diff <= 0 ? 0 : diff;
+    return data;
+
+}
+
+function fillTable(dataList) {
+    dataList.filter(function(data) { return data; }).forEach(function (data) {
+        table.insertRecord(data.server,
+            [data.issuer, data.expiry, data.remainingDays, data.valid, data.certError]
+        );
+    });
+    return table;
+}
+
+function failure(err){
+    console.error(err);
+    D.failure();
+}
+
+/**
+* @remote_procedure
+* @label Validate Association
+* @documentation This procedure is used to validate if the driver can be applied on a device during association as well as validate any credentials provided
+*/
+function validate() {
```

Comment:
This function crashes --> '	[ErrorWrapper(exc = NoneIsNotAllowedError(), loc = ('variables', 0))],'